Docs: Refactor "Environment variables" into 3 articles (Diátaxis) (#9966)
readthedocs/readthedocs.org: docs/user/environment-variables.rst

.. _Environment Variables:

Understanding environment variables
===================================

> Read the Docs allows you to define your own environment variables to be used in the build process.
> This is useful for adding build secrets such as API tokens.

Reviewer: ```suggestion
Read the Docs allows you to define your own environment variables to be used in the build process.
It also defines a set of :doc:`default environment variables</reference/environment-variables>` with information about your build.
These are useful for different purposes:

* Custom environment variables are useful for adding build secrets such as API tokens.
* Default environment variables are useful for varying your build specifically for Read the Docs or specific types of builds on Read the Docs. 
```

Reviewer: I don't feel too strongly. We use `default` in a _lot_ of places (eg. `default` values of things). I think `default` makes sense if they can be overwritten, but I don't think that's common here, so `pre-defined` is probably better. Or even `Build-specific environment variables` to be even more explicit and useful (who cares if they are pre-defined, what _are_ they?)

.. The following paragraph is difficult to balance.
.. We should ideally support environment variables in the Config File,
.. but as long as it's not supported then people can add environment variables in different ways.
.. Using the Dashboard is a good approach
.. but adding an environment variable with ``ENV=123 command --flag`` is possibly better.

> Environment variables are defined in the :term:`dashboard` interface in :menuselection:`Admin --> Environment variables`.

Reviewer: ```suggestion
Custom environment variables are defined in the :term:`dashboard` interface in :menuselection:`Admin --> Environment variables`.
```

Author: Yes please!!! :pray:

> Environment variables are configured and managed for a project's entire build process `with 2 exceptions <Environment variables and build environments>`_.

Reviewer: I think if we use a `:ref:` here it will hover :) Also the link is broken. I believe `:ref:` will also fail the build on broken links. We should almost certainly fail the build for broken _internal_ links as well.

> Environment variables are configured and managed for a project's entire build process `with 2 exceptions <Environment variables and build environments>`_.

Reviewer: ```suggestion
Environment variables are defined for a project's entire build process,
`with 2 important exceptions <Environment variables and build environments>`_.
```

Author: Thanks! Yes! Gonna fix the link as well...

Aside from storing secrets,
there are :ref:`other patterns <environment-variables:Patterns of using environment variables>` that take advantage of environment variables,

> for instance to reuse the same *monorepo* configuration in multiple documentation projects.

Reviewer: ```suggestion
like reusing the same *monorepo* configuration in multiple documentation projects.
```

> In cases where the environment variable isn't a secret,

Reviewer: ```suggestion
In cases where the environment variable isn't a secret,
like a build tool flag,
```
